### ISSUE TITLE
[Feature Request] Allow uploading files 

### ISSUE THREAD

**Issue by sarpaykent** [bakaphp/http#1]
Hello,

It would be great if we could accept uploading files (for example for blog posts) over Rest API.  That could be done with middleware or custom function before executing `$this->model->save([Post], [Allowed Params])`. I can implement that feature too, if we can decide how we should do this.

**Comment by SparoHawk** [comment]
Yes, this is something that could be done. Could you propose the structure you have in mind?

**Comment by HACKHERS** [comment]
Well, to support multiple images for single record I suggest to create table (for example: media) then we need something like:

We need a table to store media:
id <- Media id
owner_id <- ID of the record that has this media
model_name <- Full name of the class that uses this media
path(maybe url?) <- relative path of the file  
...
(More information about the file can be stored, users can extend this model to store what they need or they can use their own table.)

To handle upload and relations model behavior could help. **beforeCreate**, **beforeUpdate**, and **afterDelete** events could get some options like; directory path, post field, number of maximum/minimum images... We can get default upload directory, media model from config.

**Comment by kaioken** [comment]
@HACKHERS the scope of this project is to make api development easier for all phalcon developers so adding images is a important aspect. Now, we may need to add 2 drivers to store the image info, table or redis.

After that I'm good, @SparoHawk ?

**Comment by SparoHawk** [comment]
@kaioken Considering popular choices I'd say we need 4 drivers:
* Local Storage
* Amazon S3
* Redis
* Database

**Comment by kaioken** [comment]
this features is going to be release with canvas
